src/client/java/net/neoforged/neoforge/client/event/EntityRenderersEvent.java

```diff
@@ -227,52 +229,61 @@ public EntityRendererProvider.Context getContext() {
      */
     public static class CreateSkullModels extends EntityRenderersEvent {
         private final Map<SkullBlock.Type, Function<EntityModelSet, SkullModelBase>> skullModels;
+        private final Map<SkullBlock.Type, ResourceLocation> skullTextures;
 
         @ApiStatus.Internal
-        public CreateSkullModels(Map<SkullBlock.Type, Function<EntityModelSet, SkullModelBase>> skullModels) {
+        public CreateSkullModels(Map<SkullBlock.Type, Function<EntityModelSet, SkullModelBase>> skullModels, Map<SkullBlock.Type, ResourceLocation> skullTextures) {
             this.skullModels = skullModels;
+            this.skullTextures = skullTextures;
         }
 
         /**
-         * Registers a {@link SkullModel} for a skull block with the given {@link SkullBlock.Type}.
+         * Registers a {@link SkullModel} for a skull block with the given {@link SkullBlock.Type}, and optionally registers a skull texture to the {@link SkullBlockRenderer#SKIN_BY_TYPE} map.
          *
          * @param type          a unique skull type; an exception will be thrown if multiple mods register models
          *                      for the same type or a mod tries to register a model for a vanilla type
          * @param layerLocation the key that identifies the {@link LayerDefinition} used by the model
+         * @param skullTexture  the skull texture to put in the {@link SkullBlockRenderer#SKIN_BY_TYPE} map.
          */
-        public void registerSkullModel(SkullBlock.Type type, ModelLayerLocation layerLocation) {
-            this.registerSkullModel(type, layerLocation, SkullModel::new);
+        public void registerSkullModel(SkullBlock.Type type, ModelLayerLocation layerLocation, @Nullable ResourceLocation skullTexture) {
+            this.registerSkullModel(type, layerLocation, SkullModel::new, skullTexture);
         }
 
         /**
-         * Registers the entity model for a skull block with the given {@link SkullBlock.Type}.
+         * Registers the entity model for a skull block with the given {@link SkullBlock.Type}, and optionally registers a skull texture to the {@link SkullBlockRenderer#SKIN_BY_TYPE} map.
          *
          * @param type          a unique skull type; an exception will be thrown if multiple mods register models
          *                      for the same type or a mod tries to register a model for a vanilla type
          * @param layerLocation the key that identifies the {@link LayerDefinition} used by the model
          * @param factory       the factory to create the skull model instance, taking in the root {@link ModelPart} and
          *                      returning the model.
+         * @param skullTexture  the skull texture to put in the {@link SkullBlockRenderer#SKIN_BY_TYPE} map.
```

Comment:
```suggestion
         * @param skullTexture  the skull texture to put in the {@link SkullBlockRenderer#SKIN_BY_TYPE} map, if provided.
```
I meant add the `if provided` to each of the parameter descriptions.